pwndbg/commands/leakfind.py

```diff
@@ -0,0 +1,155 @@
+def leakfind(address=-1,page_name=None,max_offset=0x40,max_depth=0x4,stride=0x1,negative_offset=0x0):
+    if address == -1:
+        print("No starting address provided. Please run leakfind -h for more information.")
+        return
+
+    foundPages = getPage(address)
+
+    if not foundPages:
+        print("Starting address is not mapped. Please run leakfind -h for more information.")
+        return
+
+    if not pwndbg.memory.peek(address):
+        print("Unable to read from starting address. Please run leakfind -h for more information.")
+        return
+
+    max_depth=int(max_depth)
+    #Just warn the user that a large depth might be slow. Probably worth checking offset^depth < threshold. Do this when more benchmarking is established.
+    if max_depth > 8:
+        print("leakfind may take a while to run on larger depths.")
+    
+    stride = int(stride)
+    address=int(address)
+    max_offset = int(max_offset)
+    negative_offset=int(negative_offset)
+    
+    #The below map stores a map of child address->(parent_address,parent_start_address)
+    #In the above tuple, parent_address is the exact address with a pointer to the child adddress.
+    #parent_start_address is an address that a previous address pointed to.
+    #We need to store both so that we can nicely create our leak chain.
+    visitedMap = {}
+    visitedSet = set()
+    visitedSet.add(int(address))
+    addressQueue = Queue()
+    addressQueue.put(int(address))
+    depth = 0
+    timeToDepthIncrease=0
+
+    #Run a bfs
+    #TODO look into performance gain from checking if an address is mapped before calling pwndbg.memory.pvoid()
+    #TODO also check using pwndbg.memory.read for possible performance boosts.
+    while addressQueue.qsize() > 0 and depth < max_depth:
+        if timeToDepthIncrease == 0:
+            depth=depth+1
+            timeToDepthIncrease=addressQueue.qsize()
+        cur_start_addr = addressQueue.get()
+        timeToDepthIncrease-=1
+        for cur_addr in range(cur_start_addr-negative_offset,cur_start_addr+max_offset,stride):
+            try:
+                result = int(pwndbg.memory.pvoid(cur_addr))
+                if result in visitedMap:
+                    continue
+                if result in visitedSet:
+                    continue
+                visitedMap[result]=(cur_addr,cur_start_addr) #map is of form child->(parent,parent_start)
+                addressQueue.put(result)
+                visitedSet.add(result)
+            except gdb.error:
+                #That means the memory was unreadable. Just skip it if we can't read it. 
+                break
+
+    #A map of length->list of lines. Used to let us print in a somewhat nice manner.
+    outputMap = {}
+
+ 
+    for child in visitedMap:
+        childPage = getPage(child)
+        if childPage is not None:
+            if page_name is not None:
+                if not page_name in childPage.objfile:
+                    continue
```

Comment:
Just:
```python
if page_name is not None or page_name not in child_page.objfile:
    continue
```

Please note that there is a huge difference between:
```python
x = not X in Y
y = X not in Y
```

and I assume you meant `X not in Y` instead of `not X in Y`

Reply:
Fixed